crates/sui-indexer/src/lib.rs

```diff
@@ -42,26 +44,59 @@ pub type PgPoolConnection = PooledConnection<ConnectionManager<PgConnection>>;
 // TODO: placeholder, read from env or config file.
 pub const FAKE_PKG_VERSION: &str = "0.0.0";
 
+#[derive(Parser, Clone, Debug)]
+#[clap(
+    name = "Sui indexer",
+    about = "An off-fullnode service serving data from Sui protocol",
+    rename_all = "kebab-case"
+)]
+pub struct IndexerConfig {
+    #[clap(long)]
+    pub db_url: String,
+    #[clap(long)]
+    pub rpc_client_url: String,
+    #[clap(long, default_value = "0.0.0.0", global = true)]
```

Comment:
question: iirc this is for PE? should we keep the default_value to 127 since that's been the default previously or does it not matter?

Reply:
oh i see that's handled by the `default` function

Reply:
actually, do we need to set default values here if we have that `default` fn?

Reply:
You do because the default config is a function call, but see the below line 

```
            let rpc_client = new_rpc_client(config.rpc_client_url.as_str()).await?;
```
But here it looks like `rpc_client_url` has no default since we expect the user to always pass this url.

Reply:
the default() today is only used for slightly easier testing setup, this PR is mainly for PE yes

Reply:
the default values here is for clap / CLI inputs.

Reply:
Thanks, I understand now (I think). `main.rs` calls `IndexerConfig::parse()` so we'd need to set defaults on the cli args. My erroneous assumption was that `default` gets called. But instead one needs to explicitly call `IndexerConfig::default`. Appreciate everyone's responses!